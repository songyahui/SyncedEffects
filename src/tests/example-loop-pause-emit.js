"use hopscript"

var hh = require("hiphop");

var prg = <hh.module I=${{accessibility: hh.IN}} S>
  <hh.loop>
    <hh.Sequence>
      <hh.await I/>
      <hh.pause/>
      <hh.emit S/>
    </hh.Sequence>
  </hh.loop>
</hh.module>;

exports.prg = new hh.ReactiveMachine(prg, "looppauseemit");
