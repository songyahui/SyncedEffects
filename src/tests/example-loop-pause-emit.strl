module looppauseemit:

input I;
output S;

loop
	await I;
	pause;
	emit S;
end loop

end module