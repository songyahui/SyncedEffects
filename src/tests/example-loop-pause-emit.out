looppauseemit> ;
--- Output:
looppauseemit> I;
--- Output:
looppauseemit> I;
--- Output: S
looppauseemit> I;
--- Output:
looppauseemit> ;
--- Output: S
looppauseemit> ;
--- Output:
looppauseemit> ;
--- Output:
looppauseemit> I;
--- Output:
looppauseemit> ;
--- Output: S
looppauseemit> ;
--- Output:
